### PR TITLE
Add support for all algorithms in Transport::Algorithms#host_key_format

### PR DESCRIPTION
**Pull request by brbrady**
Currently, only ssh-rsa is supported; others that also come in in the format `#{cipher_name}-cert-v0[01]@openssh.com` will break currently.

example error:
```
Net::SSH::Exception: host key algorithm mismatch for signature 'ecdsa-sha2-nistp256' != 'ecdsa-sha2-nistp256-cert-v01@openssh.com'
/var/lib/gems/2.3.0/gems/net-ssh-5.2.0/lib/net/ssh/transport/kex/ecdh_sha2_nistp256.rb:88:in `send_kexinit'
```

### CODE REVIEW

**Comment by mcpherrinm** [comment]
Hi, any interest in reviewing/merging this?  I'm seeing trouble with ssh-ed25519-cert-v01@openssh.com and Net::Ssh

**Comment by mcpherrinm** [comment]
@mfazekas Is there anything we can do to help get this reviewed and merged?

**Comment by mfazekas** [comment]
@mcpherrinm  have you tried this branch? Did you have an issue before and did this branch solved it?

File: lib/net/ssh/transport/algorithms.rb [hunk 1]
```diff
@@ -197,8 +197,8 @@ def initialized?
 
         def host_key_format
           case host_key
-          when "ssh-rsa-cert-v01@openssh.com", "ssh-rsa-cert-v00@openssh.com"
-            "ssh-rsa"
+          when /^([a-z0-9-]+)-cert-v\d{2}@openssh.com$/
+            $1 # rubocop:disable Style/PerlBackrefs
```
Comment (mfazekas) [inline lib/net/ssh/transport/algorithms.rb]: I'd say disable the PerlBackref check globally

**Comment by mcpherrinm** [comment]
Yes, I have tried this branch and it fixes the exception I'm hitting,  #<Net::SSH::Exception: host key algorithm mismatch for signature 'ssh-ed25519' != 'ssh-ed25519-cert-v01@openssh.com'>

**Comment by mcpherrinm** [comment]
Just want to follow up, can we get this merged?

We've been using @brbrady's fork but would prefer to use mainline net-ssh.

File: lib/net/ssh/transport/algorithms.rb (hunk 1 above)
Comment (baweaver) [reply lib/net/ssh/transport/algorithms.rb]: This could also be substituted for `$MATCH` to use the english variant.

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -1,113 +1,140 @@
 # This configuration was generated by
 # `rubocop --auto-gen-config`
-# on 2016-12-11 13:27:14 +0100 using RuboCop version 0.46.0.
+# on 2019-06-06 14:12:57 -0700 using RuboCop version 0.54.0.
```
Comment (fwininger) [inline .rubocop_todo.yml]: Please rebase your branch with master before. We currently use the latest version of rubocop.

File: test/transport/test_algorithms.rb [hunk 3]
```diff
@@ -377,10 +384,10 @@ def assert_exchange_results
       assert_equal :server_hmac, transport.server_options[:hmac]
     end
 
-    def algorithms(algrothms_options={}, transport_options={})
+    def algorithms(algorithms_options={}, transport_options={})
```
Comment (fwininger) [inline test/transport/test_algorithms.rb]: :+1:

**Comment by mfazekas** [comment]
> Just want to follow up, can we get this merged?
> 
> We've been using @brbrady's fork but would prefer to use mainline net-ssh.

Sure this can be merged, but we'd need to address what's brought up in review.

**Comment by mfazekas** [comment]
Merged as: #721

**Pull request closed without merging**